svm/src/account_loader.rs

```diff
@@ -37,9 +37,15 @@ use {
 // for the load instructions
 pub(crate) type TransactionRent = u64;
 pub(crate) type TransactionProgramIndices = Vec<Vec<IndexOfAccount>>;
-pub type TransactionCheckResult = (transaction::Result<()>, Option<NoncePartial>, Option<u64>);
+pub type TransactionCheckResult = Result<CheckedTransactionDetails>;
 pub type TransactionLoadResult = Result<LoadedTransaction>;
 
+#[derive(PartialEq, Eq, Debug, Clone)]
+pub struct CheckedTransactionDetails {
+    pub nonce: Option<NoncePartial>,
+    pub lamports_per_signature: Option<u64>,
```

Comment:
Can these items be None even when the transaction succeeded? Perhaps we could clean these too.

Reply:
`nonce` will need to remain `None` since it dictates whether a transaction used a durable nonce or not. But yes, `lamports_per_signature` will never be `None` and I will clean that up in a follow-up

Reply:
> But yes, `lamports_per_signature` will never be `None` and I will clean that up in a follow-up

That will be great!